docs/1_python.md

```diff
@@ -0,0 +1,197 @@
+# Oras Python User Guide
+
+Oras Python will allow you to run traditional push and pull commands for artifacts,
+or generate a custom client. This small user guide will walk you through these various
+steps, and please open an issue if functionality is missing.
+
+## Installation 
+
+### Pypi
+
+The module is available in pypi as [oras](https://pypi.org/project/oras/)
+and you can install as follows:
+
+```bash
+$ pip install oras
+``` 
+   
+You can also clone the repository and install locally:
+
+```bash
+$ git clone https://github.com/oras-project/oras-py
+$ cd oras-py
+$ pip install .
+``` 
+
+Or in development mode:
+
+```bash
+$ pip install -e .
+```
+
+Development mode means that the install is done from where you've cloned the library,
+so any changes you make are immediately "live" for testing.
+
+### Docker Container
+
+We provide a [Dockerfile](https://github.com/oras-project/oras-py/blob/main/Dockerfile) to build a container with the client.
+
+```bash
+$ docker build -t oras-py .
+
+$ docker run -it oras-py                                                                                                                   
+# which oras-py
+/opt/conda/bin/oras-py
+```
+
+## Registry
+
+You should see [supported registries](https://oras.land/implementors/#docker-distribution) or if you
+want to deploy a local testing registry (without auth), you can do:
+
+
+```bash
+$ docker run -it --rm -p 5000:5000 ghcr.io/oras-project/registry:latest
+```
+
+To test token authentication, you can either [set up your own auth server](https://github.com/adigunhammedolalekan/registry-auth)
+or just use an actual registry. The most we can do here is set up an example that uses basic auth.
+
+
+```bash
+# This is an htpassword file, "b" means bcrypt
+htpasswd -cB -b auth.htpasswd myuser mypass
+```
+
+> :warning: The server below will work to login with basic auth, but you won't be able to issue tokens.
+
+```bash
+# And start the registry with authentication
+docker run -it --rm -p 5000:5000 \
+    -v $(pwd)/auth.htpasswd:/etc/docker/registry/auth.htpasswd \
+    -e REGISTRY_AUTH="{htpasswd: {realm: localhost, path: /etc/docker/registry/auth.htpasswd}}" \
+    ghcr.io/oras-project/registry:latest
+```
+
+## Command Line
+
+### Login
+
+Once you create (or already have) a registry, you will want to login. You can do:
+
+```bash
+$ oras-py login -u myuser -p mypass localhost:5000
+
+# or localhost (insecure)
+$ oras-py login -u myuser -p mypass -k localhost:5000 --insecure
```

Comment:
Note that `--insecure` means skipping certificate verification for TLS while `--plain-http` means `HTTP` instead of `HTTPS` for connections.

Are we using the correct flags?

Reply:
BTW, what is the full option name for `-k`? or is it a typo? Note that `-k` is not there at line 83.

Reply:
@shizhMSFT it's likely not, and this is noted in #5. This would be a separate issue from the documentation here, that when fixed we'd update it here too. The current scope of work for this PR is to get the current docs deployed on oras.land.

Reply:
`-k` is insecure https://github.com/oras-project/oras-py/blob/0521bd85a1b544cf3bb3132aee9e7f1d3b16f29c/oras/cli/__init__.py#L133-L140

but should be `--keep-old-files` with https://github.com/oras-project/oras/blob/3149134ede7e0e065a34e762b9ff444d82803305/cmd/oras/pull.go#L73 (it looks like a typo).

Reply:
I can fix that here, give me a minute.

Reply:
okay -k is fixed, new issue opened for clarifying the flags https://github.com/oras-project/oras-py/issues/29